Made project report changes

shriram-sg opened the pull request:
1. Changed about page to include details about python first.
2. Removed explanation from softwares used page and moved school syllabus related items up.
3. Removed feasibility study subsections from ToC

shriram-sg committed:
add changed latex and pdf files

anand2312 commented on latex-report/main.tex:
```diff
@@ -188,35 +188,35 @@
   The aim of the project, Sudoku Webapp, is to provide an aesthetically rich interface for sudoku enthusiasts to enjoy sudoku puzzles.\newline
 
-  A website with 3 webpages: home page, sudoku page and leaderboard page, is the frontend of the project with which the user interacts. The webpages are made using Django, HTML, JavaScript and a CSS framework called Tailwind. A MySQL database is the primary data store, and is used to store the leaderboard data, and the sudoku game boards that are served on the site.\newline
+  A website with 3 webpages - home page, sudoku page and leaderboard page - is the frontend of the project with which the user interacts. The entire backend is handled using python and the django framework. The sudoku game boards are generated using a combination of custom classes and functions as well as functions from the random module in python. These game boards are passed onto the website us ...
```
Capitalize `Python` and `Django` here I'd say; and if possible everywhere else they're used. (Not very high priority though)

shriram-sg replied:
Done

anand2312 commented on latex-report/main.tex:
```diff
@@ -641,14 +641,14 @@
             count += 1
 
-    def __setitem__(self, index: int, val: typing.Union[int, None]):  # noqa: ANN204
+    def __setitem__(self, index: int, val: typing.Optional[int]=None):  # noqa: ANN204
```
Nice, using the Optional type hint is better 👍🏻 
If you do end up editing this file again, add [spaces around the `=` for the default](https://www.python.org/dev/peps/pep-3107/#parameters) (Note that if there are no type-hints, no space is expected there)

shriram-sg replied:
I'll make the change, but lets not print it out.

anand2312 reviewed (approved):
LGTM, you can decide if you want to make these changes (rather minor, so I don't mind if you don't)

shriram-sg committed:
make minor changes

1. Capitalization of 'Python' and 'Django'
2. Space in the 'typing.Optional' type hint

anand2312 reviewed (approved):
LGTM, merge it 👍

shriram-sg committed:
update the general pdf
